File: WarehouseManagementWeb.Application/Dto/Output/Client/ClientListByStatusOutput.cs
namespace WarehouseManagementWeb.Application.Dto.Output.Client
{
    /// <summary>
    /// Класс выходной модели клиентов разделенных по статусам.
    /// </summary>
    public class ClientListByStatusOutput
    {
        /// <summary>
        /// Список активных клиентов.
        /// </summary>
        public IEnumerable<ClientOutput>? ActiveClients { get; set; }

        /// <summary>
        /// Список клиентов находящихся в архиве.
        /// </summary>
        public IEnumerable<ClientOutput>? ArchivedClients { get; set; }
    }
}

File: warehouse-management-web-angular/src/app/directory/client/models/output/client-list-by-status-output.ts
import { ClientOutput } from './client-output';

/**
 * Класс выходной модели клиентов разделенных по статусам.
 */
export class ClientListByStatusOutput {
  /**
   * Список актвиных клиентов.
   */
  activeClients: ClientOutput[] = [];

  /**
   *Список клиентов находящихся в архиве.
   */
  archivedClients: ClientOutput[] = [];
}
